### ISSUE TITLE
Vert.x ConfigMap quickstart installed from new launcher builds - fails to run in production

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#2988]
Steps to recreate:
* In the app launcher, create a new app with Vert.X + ConfigMap
* Observe build pipeline completes successfully
* Attempt to run the app on stage - observe this error:

In the stage log:
```
Apr 11, 2018 2:14:32 AM io.vertx.config.impl.ConfigRetrieverImpl
  | SEVERE: Error while scanning configuration
  | io.vertx.core.impl.NoStackTraceThrowable: Access denied to configmap or secret in namespace ldimaggi-stage: app-config
```

![screenshot from 2018-04-10 22-14-23](https://user-images.githubusercontent.com/642621/38593071-8817343e-3d0d-11e8-83ae-26a8533dd10f.png)

**Comment by ldimaggi** [comment]
Eclipse Vert.x 3.4.2.redhat-009 (RHOAR)

**Comment by gastaldi** [comment]
@ldimaggi please retest. The booster was updated this week

**Comment by ldimaggi** [comment]
Same problem is still there:

![screenshot from 2018-04-18 23-04-10](https://user-images.githubusercontent.com/642621/38969174-e2fa5aa2-435c-11e8-9785-5c59252e061b.png)

**Comment by ldimaggi** [comment]
The test is failing in production - it looks good in prod-preview. Thx!

**Comment by gastaldi** [comment]
Try in production now, the catalog was released

**Comment by gastaldi** [comment]
What is the repository that was created? Do you know if it's different from the one you tested in prod-preview?

**Comment by ldimaggi** [comment]
Sorry for delay in getting back to you - the repo is here:  https://github.com/osiotestmachine/testapr191524190994388

**Comment by gastaldi** [comment]
This is not a Configmap booster. Are you sure it is the right repo?

**Comment by ldimaggi** [comment]
Sorry - this is a Vert.X/ConfigMap repo:
https://github.com/osiotestmachine/april22

**Comment by gastaldi** [comment]
Ah, it is still using the old code because https://github.com/openshiftio/saas-launchpad/pull/94 wasn't merged yet. @joshuawilson is supposed to merge https://github.com/openshiftio/saas-openshiftio/pull/783  first

**Comment by ldimaggi** [comment]
This is still failing:
![screenshot from 2018-04-30 16-13-47](https://user-images.githubusercontent.com/642621/39447980-821822c6-4c91-11e8-8090-05ff1751da69.png)

What specific steps are you following? The steps defined in the quickstart's README?

**Comment by l0rd** [comment]
@slemeur @ldimaggi not sure the labels `area/che` `team/che` are appropriate here or you should be more precise about what is not working in Che.

**Comment by ibuziuk** [comment]
@ldimaggi please, explain your last comment ? the issue was about running app on staging which as I understand now works, right?  If there is a problem with running app in che and making curl request as defined in readme we probably need a new issue for this

**Comment by ibuziuk** [comment]
Some notes from Che side. As I understand `area/che` was added due to the fact that running the app from che results in the following error:

![image](https://user-images.githubusercontent.com/1461122/39535354-f9ccf2c6-4e33-11e8-9cf8-1e6595d0b0cf.png)

Also, as @ldimaggi mentioned in https://github.com/openshiftio/openshift.io/issues/2988#issuecomment-385515141  `curl http://localhost:8080/api/greeting` from che terminal results in `{"content":"no config map"}`

However, if you run the app locally you will get the very same problem, so Che works pretty consistent. It looks like there is some problem with quickstart itself and it worth creating an issue for this particular quickstart.

The easiest workaround for Che would be updating `run` command on stack level if there would not be fix provided from the quickstart side:

```
export VERTX_CONFIG_PATH="${current.project.path}/app-config.yml"
scl enable rh-maven33 'mvn compile vertx:run -f ${current.project.path}' 
```
A part from the fact that this makes stack project dependent it might have unpredictable effects on other vertx quickstarts though.

Basically, it seems that config path is not configured by default in the app which makes setting `VERTX_CONFIG_PATH` a requirement. However, after skimming vetx docs [1] it looks like it is possible to define config path on app level.

[1] https://vertx.io/docs/vertx-config/java/

**Comment by gastaldi** [comment]
Looping @cescoffier into this

**Comment by cescoffier** [comment]
Vert.x 3.5.0+ uses `conf/config.json` as default location for the configuration. It also supports YAML (or whatever format you want) if the format has been added to the application.

**Comment by cescoffier** [comment]
I would also add that this booster has been modified to be runnable using openshift.io, initially it was different so yes, this is not how we recommend using configuration anymore.

**Comment by ibuziuk** [comment]
@cescoffier @gastaldi quickstart used on osio uses `3.4.2.redhat-009` version and after creating `conf/config.yaml` locally I have the very same problem with `Unable to read the configuration` configuration

**Comment by cescoffier** [comment]
@ibuziuk Yes, the changes have been made in 3.5.0. Fortunately, a new version is under release (waiting for MRRC sync).

**Comment by ibuziuk** [comment]
@cescoffier @gastaldi am I correct that the plan is the following:
- updating vertx version in quickstart
- changing location of config file in quickstart 
- updating version of quickstart used on osio  ?

**Comment by maxandersen** [comment]
@cescoffer when running in the IDE why would you want to read the possible stale values from the configmaps ?

**Comment by ibuziuk** [comment]
@ldimaggi @rhopp could you please test if it is fixed on prod-preiew / prod with a new verison of booster ?

**Comment by maxandersen** [comment]
> @maxandersen I don't understand the question about stale values? Do you mean an outdated config map?

when you run in IDE which kube config map do you expect to fetch ?

**Comment by ibuziuk** [comment]
@cescoffier could you please point to PR with update ? does not seem to be working on prod ?

**Comment by ibuziuk** [comment]
@maxandersen @slemeur @cescoffier just tested locally - the version of quickstart [1] used on osio does not work locally - see https://github.com/openshiftio/openshift.io/issues/2988#issuecomment-386035987

However, I see that version of vertx was updated:

` <vertx.version>3.5.1.redhat-003</vertx.version>
`
But, there is still a bug in the quickstart[1]

As you can see `app-config.yml` is still located in the root of project. However, if I place config to `conf/config.yml` it still does not work locally & configuration is not picked-up automatically :-( 

P.S. flag `-Drtx.disableDnsResolver=true` does not matter here 

[1] https://github.com/ibuziuk/app-test-1asdasdasaaaaaaaaaaaa

**Comment by ibuziuk** [comment]
Upstream quickstart issue - https://github.com/openshiftio-vertx-boosters/vertx-configmap-booster-redhat/issues/14

**Comment by slemeur** [comment]
@cescoffier Unfortunately, the issue is not fixed in the booster itself and we are blocked here. Any suggestion on openshiftio-vertx-boosters/vertx-configmap-booster-redhat#14 ?

**Comment by cescoffier** [comment]
@slemeur Can you confirm that the issue is when running `vertx:run` in che?

**Comment by slemeur** [comment]
Yes that's the command we are using:
`scl enable rh-maven33 'mvn compile vertx:run -f ${current.project.path} -Dvertx.disableDnsResolver=true'` 

![screen shot 2018-05-05 at 10 03 14](https://user-images.githubusercontent.com/1636769/39661050-c698adae-504b-11e8-94b3-6517c9e1a054.png)

**Comment by slemeur** [comment]
Awesome, thanks @cescoffier ! 
Let us know when we can test.

**Comment by cescoffier** [comment]
@slemeur Do you need it in the launcher to test or if I push to master you would be able to check?

**Comment by cescoffier** [comment]
@slemeur changes have been made in the `master` branch of the booster. Release in progress (but it can take some time before it's merged in the launcher.

**Comment by cescoffier** [comment]
PR on launcher openeded: https://github.com/fabric8-launcher/launcher-booster-catalog/pull/230

**Comment by slemeur** [comment]
Wonderful ! I tested and it works in Che 👍 Thanks @cescoffier 

@ibuziuk : The change means that we need should rebuild the stacks for vert.x one more time. Then we should be fine.

**Comment by slemeur** [comment]
Ok great !

**Comment by ibuziuk** [comment]
PR for stack rebuilding - https://github.com/redhat-developer/che-dockerfiles/pull/36

**Comment by gastaldi** [comment]
PR in booster-catalog merged and released to production

**Comment by ibuziuk** [comment]
PR for rebuilding stacks is also merged, so looks like we can close this issue

**Comment by slemeur** [comment]
Tested, fixed in Che.
Thanks @cescoffier , @ibuziuk , @gastaldi !
